src/btree/bt_delete.c

```diff
@@ -194,11 +194,19 @@ __wti_delete_page(WT_SESSION_IMPL *session, WT_REF *ref, bool *skipp)
      */
     WT_ERR(__wt_page_parent_modify_set(session, ref, false));
 
-    WT_ERR(__wt_calloc_one(session, &ref->page_del));
-
-    /* History store truncation is non-transactional. */
-    if (!WT_IS_HS(session->dhandle))
+    /*
+     * History store truncation is non-transactional, and fast truncate in history store applies
+     * only to globally visible pages. Therefore, ensure `page_del` remains null. (`page_del` being
```

Comment:
Reading this was a bit confusing with the new code below. The code reads to me as we only allocate page_del when the btree is not history store. Personally I would like to have a comment describing the code and then have an additional paragraph explaining why not history store. E.g.
```
Allocation page_del structure on pages in tables except from the history store. 

page_del being null indicates that the fast truncated page is globally visible. History store truncation is non-transactional and therefore is usually fine with the only exception being selective backup. During recovery of a selective backup, tables that are not included in the backup are truncated. This means that it is possible for non-globally visible pages to be truncated.
```

Something like that, the aim is give a general gist of my idea, you can modify as much as you want.

Reply:
Good one, updated in [24c91f3](https://github.com/wiredtiger/wiredtiger/pull/11190/commits/24c91f3ab0445f7047b72fc198c8928b973cdb0e). let me know what you think.